### ISSUE TITLE
Error with returning BIGINT (20) ID's

### ISSUE THREAD

**Issue by jimmyjamieson** [balderdashy/sails-mysql#265]
I'm trying to display a table in sails from a mysql table using the mysql-adapter The ID's in the table are in the following format

```
ID
14482773051001000000
14482773051001000001
14482773051001000002
```

But sails is spitting out every ID like

```
14482773051001000000
14482773051001000000
14482773051001000000
```

Basically replacing all the last digits with 0.

I've tried autoPK false and setting size to 64 for a bigint. It works if I use my seed on development where the ID's are incremented 1,2,3,4,etc it works fine as the ID's are aren't the 20 numbers long.

My attributes are currently set like so:
```
 autoPK: false,
 schema: true,

  attributes: {
    id: {
  type: 'integer',
      size: 64,
      unique: true,
      autoIncrement:true,
      primaryKey: true
    },
```
Now, I'm stuck as I can't seen to get the correct data in the API, which in turn breaks my front-end.
Is this a known bug? Is there a fix? Is there something I can do to fix the issue myself?

**Comment by jimmyjamieson** [comment]
I've ran an experiment and it seems it's only keeping the first 16 numbers, then replacing the last 4 with 0000.

Database is set at bigint(20) :/ Obviously this is a big problem.

**Comment by jimmyjamieson** [comment]
So JS's maximum number is 16 digits. This seems to be the issue. There's node packages such as https://www.npmjs.com/package/big-integer 

Could we do something similar here?

**Comment by jimmyjamieson** [comment]
Answer was to turn on supportBigNumbers: true via the MysqlAdapter

stagingMysql: {
    adapter: 'sails-mysql',
    host: '192.168.x.x',
    user: 'username',
    password: 'password',
    database: 'myDb',
    supportBigNumbers: true
  },

Thanks to @japel https://github.com/japel on gitter

**Comment by japel** [comment]
np m8

**Comment by sailsbot** [comment]
Thanks for posting, @jimmyjamieson.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
